include/gnc/environment.hpp

```diff
@@ -9,38 +40,145 @@
 namespace gnc {
 namespace env {
 
-/** @fn earth_attitude
- *  @param[in]  t Time in seconds since the PAN epoch.
- *  @param[out] q Quaternion to transform vectors from ECI to ECEF.
- *  @{ */
-void earth_attitude(double t, lin::Vector4d &q);
-void earth_attitude(double t, lin::Vector4f &q);
-/** @} */
-
-/** @fn earth_angular_rate
- *  @param[in]  t Time in seconds since the PAN epoch.
- *  @param[out] w Angular rate to transform vectors from ECI to ECEF.
- *  @{ */
-void earth_angular_rate(double t, lin::Vector3d &w);
-void earth_angular_rate(double t, lin::Vector3f &w);
-/** @} */
-
-/** @fn sun_vector
- *  @param[in]  t Time in seconds since the PAN epoch.
- *  @param[out] s Normalized vector from Earth the Sun in the ECI frame.
- *  @{ */
-void sun_vector(double t, lin::Vector3f &s);
-void sun_vector(double t, lin::Vector3d &s);
-/** @} */
-
-/** @fn magnetic_field
- *  @param[in]  t Time in seconds since the PAN epoch.
- *  @param[in]  r Position in the ECEF frame with units of meters.
- *  @param[out] b Magnetic field in the ECEF frame with units of Tesla.
- *  @{ */
-void magnetic_field(double t, lin::Vector3d const &r, lin::Vector3f &b);
-void magnetic_field(double t, lin::Vector3d const &r, lin::Vector3d &b);
-/** @} */
+/** @brief Determines earth's current attitude represented as a quaternion.
+ *
+ *  @param[in]  t          Time in seconds since the PAN epoch.
+ *  @param[out] q_ecef_eci Earth's attitude represented as a quaternion.
+ * 
+ *  The returned quaternion represents the rotation transforming from ECI to ECEF.
+ *  This representation will be most accurate around the PAN epoch.
+ * 
+ *  @sa env::earth_attitude(double, lin::Vector4d &)
+ * 
+ *  @ingroup environment
+ */
+void earth_attitude(double t, lin::Vector4d &q_ecef_eci);
+
+/** @brief Determines earth's current attitude represented as a quaternion.
+ *
+ *  @param[in]  t          Time in seconds since the PAN epoch.
+ *  @param[out] q_ecef_eci Earth's attitude represented as a quaternion.
+ * 
+ *  The returned quaternion represents the rotation transforming from ECI to ECEF.
+ *  This representation will be most accurate around the PAN epoch.
+ * 
+ *  @sa env::earth_attitude(double, lin::Vector4f &)
+ * 
+ *  @ingroup environment
+ */
+void earth_attitude(double t, lin::Vector4f &q_ecef_eci);
+
+/** @brief Determines Earth's angular rate.
+ *
+ *  @param[in]  t     Time in seconds since the PAN epoch.
+ *  @param[out] w_eci Earth's angular rate (units of radians per second).
+ *
+ *  @internal
+ *  Currently, the time input is extraneous as this function only looks at Earth's
+ *  rate about the z-axis and doesn't model precession/nutation.
```

Comment:
[Polar motion](https://en.wikipedia.org/wiki/Polar_motion) not precession/nutation